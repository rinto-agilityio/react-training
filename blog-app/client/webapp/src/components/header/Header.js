import React, { memo } from 'react';
import { Link } from 'react-router-dom';

import { Query } from 'react-apollo';

//query
import { LOGGED_USER } from '../../graphql/author/queries';

//components
import Image from '../commons/Image';
import Dropdown from '../commons/Dropdown';

//styles
import './HeaderStyle.css';

const Header = () => {
  const handleLogout = (event, client) => {
    event.stopPropagation()
    client.resetStore();
    localStorage.removeItem('userLoged');
  }

  return (
    <Query
      query={LOGGED_USER}
      fetchPolicy={'cache-only'}
    >
      {({ data, client }) => {
        const { loggedUser } = data
        return (
          <header className="header">
            <div className='logo-wrap'>
              <h1 className="brand">
                <Link to='/' className="brand-link">
                  <div className="brand-name">
                    <p>Blog App</p>
                  </div>
                </Link>
              </h1>
            </div>
            <div className='login-bar'>
              {loggedUser ?
                (
                  <>
                    <p className='account-name'>
                      {loggedUser.name}
                    </p>
                  </>
                )
                :
                (
                  <Link className="hover-decoration" to={'/login'}>
                    LOGIN
                  </Link>
                )
              }
              <Image avarta={loggedUser && loggedUser.avatar}/>
              <Dropdown
                items={[
                  <Link
                    onClick={(event)=>handleLogout(event, client)}
                  >
                    Log Out
                  </Link>
                ]}
              />
            </div>
          </header>
        )
      }}
    </Query>
  )
}

export default memo(Header);
