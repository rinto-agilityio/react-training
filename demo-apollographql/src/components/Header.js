import React from 'react'

import {
  HeaderWrapper,
  SiteTitle,
  SiteTitleLink,
} from './Header.style'

const Header = () => (
  <HeaderWrapper>
    <SiteTitle>
      <SiteTitleLink to="/" title="React-hook demo">
        GraphQL & React
      </SiteTitleLink>
    </SiteTitle>
  </HeaderWrapper>
)

export default Header
