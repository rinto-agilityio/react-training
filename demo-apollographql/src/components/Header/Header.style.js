import styled from 'styled-components'

// Components
import Grid from '@material-ui/core/Grid'
import Paper from '@material-ui/core/Paper'
import { Link } from 'react-router-dom'

export const Wrapper = styled(Grid)({
  marginBottom: '1rem',
})

export const Container = styled(Paper)({
  borderRadius: 0,
  textAlign: 'center',
})

export const StyledLink = styled(Link)({
  textDecoration: 'none',
})
