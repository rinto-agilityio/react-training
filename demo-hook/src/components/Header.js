import React from 'react'
import styled from 'styled-components'
import { Link } from 'react-router-dom'

// Themes
import { SiteThemeContext } from '../contexts/ThemeContext'

// Components
import LampIcon from './Icons/Lamp'

const HeaderWrapper = styled.nav`
  padding: 1rem;
  background-color: ${({ theme }) => theme.navBgColor };
  border-bottom: 2px solid ${({ theme }) => theme.navBorderColor};
  display: grid;
  grid-template-columns: auto auto;
`
const SiteTitle = styled.h2`
  margin: 0;
`
const SiteTitleLink = styled(Link)`
  text-decoration: none;
`
const IconWrapper = styled.div`
  cursor: pointer;
  display: inline-block;
  justify-self: right;
`

const Header = () => (
  <HeaderWrapper>
    <SiteTitle>
      <SiteTitleLink to="/" title="React-hook demo">
        React-hook Demo
      </SiteTitleLink>
    </SiteTitle>
    <SiteThemeContext.Consumer>
      {({ currentTheme, handleThemeChange }) => (
        <IconWrapper onClick={handleThemeChange}>
          <LampIcon isOn={currentTheme === 'dark'} />
        </IconWrapper>
      )}
    </SiteThemeContext.Consumer>
  </HeaderWrapper>
)

export default Header
