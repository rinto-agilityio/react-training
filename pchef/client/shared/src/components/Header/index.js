// Libs
import React, { useState } from 'react'
import { View, Text, Image, TouchableOpacity } from 'react-native'
import { Avatar, Provider, Menu } from 'react-native-paper'

// Common components
import Icon from '../Icon'
import Loading from '../Loading'
import Error from '../Error'
import Button from '../Button'

// Themes
import { COLORS, FONTS } from '../../themes'

// Styles
import styles from './styles'

// Constants
import { URL } from '../../constants/index'

type Props = {
  data: {
    user: {
      avatar: string,
    }
  },
  type?: string,
  onPressLogo?: () => void,
  onPressCategoryIcon?: () => void,
  onDirectTo: (id: string) => void,
}

const Header = ({
  type = 'primary',
  onPressLogo,
  onPressCategoryIcon,
  data = {
    user: {
      avatar: '',
    },
  },
  onDirectTo,
}: Props) => {
  const [visible, setVisible] = useState(false)
  const { user: { avatar } } = data

  return (
    <View style={[styles.wrapHeader, styles[`${type}Container`]]}>
      <View style={styles.container}>
        <Icon
          name="apps"
          size={FONTS.fontSize.extraExtraLarge}
          onPress={onPressCategoryIcon}
          color={COLORS.white}
          wrapperIconStyle={styles.wrapIcon}
        />
        <View style={styles.wrapHeaderContent}>
          <View>
            <TouchableOpacity onPress={onPressLogo} style={styles.logo}>
              <View style={[styles.imageWrapper, styles[`${type}ImageWrapper`]]}>
                <Image
                  source={require('../../assets/images/logo.png')}
                  style={[styles.image, styles[`${type}Image`]]}
                />
              </View>
              <Text style={styles.text}>Cooking</Text>
            </TouchableOpacity>
          </View>
          <View style={styles.wrapUserInfo}>
            <Button
              onPress={() => onDirectTo(URL.CREATE_RECIPE.PATH)}
              title={URL.CREATE_RECIPE.TITLE}
              buttonStyle={styles.createRecipeBtn}
            />
            <Provider>
              <Menu
                style={styles.avataDropdown}
                visible={visible}
                onDismiss={() => setVisible(false)}
                anchor={
                  (
                    <TouchableOpacity onPress={() => setVisible(true)} style={styles.logo}>
                      <Avatar.Image
                        style={styles.pipelineItemImage}
                        size={40}
                        source={{
                          uri: (
                            avatar
                              ? avatar
                              : 'https://www.pngkey.com/png/full/115-1150152_default-profile-picture-avatar-png-green.png'
                          ),
                        }}
                      />
                    </TouchableOpacity>
                  )
                }
              >
                <Menu.Item onPress={() => onDirectTo(URL.PROFILE.PATH)} title={URL.PROFILE.TITLE} />
                <Menu.Item onPress={() => onDirectTo(URL.WISH_LIST.PATH)} title={URL.WISH_LIST.TITLE} />
                <Menu.Item onPress={() => onDirectTo(URL.LOG_OUT.PATH)} title={URL.LOG_OUT.TITLE} />
              </Menu>
            </Provider>
          </View>
        </View>
      </View>
    </View>
  )
}

Header.defaultProps = {
  type: 'primary',
  onPressLogo: () => {},
  onPressCategoryIcon: () => {},
}

export default Header
