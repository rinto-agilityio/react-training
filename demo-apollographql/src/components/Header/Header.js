import React from 'react'
import PropTypes from 'prop-types'

// Components
import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography'
import { Wrapper, Container, StyledLink } from './Header.style'

const Header = ({ siteTitle }) => (
  <Wrapper container>
    <Grid item xs={12}>
      <Container>
        <Typography component="h1" variant="h3">
          <StyledLink
            to="/"
            title="React-hook demo"
          >
            {siteTitle}
          </StyledLink>
        </Typography>
      </Container>
    </Grid>
  </Wrapper>
)

Header.defaultProps = {
  siteTitle: 'GraphQL & React'
}

Header.propTypes = {
  siteTitle: PropTypes.string
}

export default Header
