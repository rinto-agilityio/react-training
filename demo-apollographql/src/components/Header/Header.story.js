import React from 'react'
import { storiesOf } from '@storybook/react'
import { MemoryRouter } from 'react-router-dom'

// Components
import Header from './index'

storiesOf('Layout', module)
  .add('Header', () => (
    <MemoryRouter>
      <Header />
    </MemoryRouter>
  ))
