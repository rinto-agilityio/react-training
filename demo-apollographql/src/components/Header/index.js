import React from 'react'

// Components
import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography'
import { Wrapper, Container, StyledLink } from './Header.style'

const Header = () => (
  <Wrapper container>
    <Grid item xs={12}>
      <Container>
        <Typography component="h1" variant="h3">
          <StyledLink
            to="/"
            title="React-hook demo"
          >
            GraphQL & React
          </StyledLink>
        </Typography>
      </Container>
    </Grid>
  </Wrapper>
)

export default Header
