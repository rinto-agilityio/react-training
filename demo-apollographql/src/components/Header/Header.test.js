import React from 'react'
import { mount } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'

// Components
import Header from './index'
import { StyledLink } from './Header.style'

const appTitle = 'GraphQL & React'

describe('Components', () => {
  describe('<Header />', () => {
    it(`Render correct app title`, () => {
      const wrapper = mount(
        <MemoryRouter>
          <Header />
        </MemoryRouter>
      )

      expect(wrapper.find(StyledLink).filter({ to: '/'}).render().text()).toEqual(appTitle)
    })

    it('Render correct snapshot', () => {
      const tree = renderer
      .create(
        <MemoryRouter>
          <Header />
        </MemoryRouter>
      )
      .toJSON()

      expect(tree).toMatchSnapshot()
    })
  })
})
