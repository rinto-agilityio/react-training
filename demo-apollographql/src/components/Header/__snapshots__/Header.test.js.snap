// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components <Header /> Render correct snapshot 1`] = `
<div
  className="MuiGrid-container-1 sc-bdVaJa fEhXQk"
>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
  >
    <div
      className="MuiPaper-root-98 MuiPaper-elevation2-102 MuiPaper-rounded-99 sc-bwzfXH dAuxoL"
    >
      <h1
        className="MuiTypography-root-125 MuiTypography-h3-139"
      >
        <a
          className="sc-htpNat llgiPg"
          href="/"
          onClick={[Function]}
          title="React-hook demo"
        >
          GraphQL & React
        </a>
      </h1>
    </div>
  </div>
</div>
`;
